go/tools/microbench/results.go

```diff
@@ -115,16 +116,51 @@ func MergeMicroBenchmarkDetails(currentMbd, lastReleaseMbd MicroBenchmarkDetails
 	return compareMbs
 }
 
-// ReduceSimpleMedian reduces a MicroBenchmarkDetailsArray by merging
+// ReduceSimpleMedianByName reduces a MicroBenchmarkDetailsArray by merging
 // all MicroBenchmarkDetails with the same benchmark name into a single
 // one. The results of each MicroBenchmarkDetails correspond to the median
 // of the merged elements.
-func (mbd MicroBenchmarkDetailsArray) ReduceSimpleMedian() (reduceMbd MicroBenchmarkDetailsArray) {
+func (mbd MicroBenchmarkDetailsArray) ReduceSimpleMedianByName() (reduceMbd MicroBenchmarkDetailsArray) {
 	sort.SliceStable(mbd, func(i, j int) bool {
-		return mbd[i].Name < mbd[j].Name
+		return mbd[i].PkgName < mbd[j].PkgName && mbd[i].Name < mbd[j].Name
```

Comment:
Filtering by both package name and name introduces duplicates in the `reduceMbd` table. I think this is due to the `mbd` slice not being sorted correctly after the call to `sort.SliceStable`, leading to algorithmic errors when we do the for loop on line 134.

```go
for j = i; j < len(mbd) && mbd[i].Name == mbd[j].Name; j++ {
   ...
}
```

**Filtering by package name and name:**
![image](https://user-images.githubusercontent.com/35779988/117300496-e28b7c00-ae79-11eb-9aac-b6cff05f3129.png)

**Filtering by name:**
![image](https://user-images.githubusercontent.com/35779988/117300615-06e75880-ae7a-11eb-91ac-b9d3092342e6.png)